### PR TITLE
B-283: fail on bad credentials

### LINKED ISSUE #283: Running plan or apply without setting the provider should return an error
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```shell
$ terraform version
Terraform v1.2.2
on darwin_arm64
+ provider registry.terraform.io/opennebula/opennebula v0.5.0
```

### Issue

Terraform shows a plan even if the credentials are wrong:

```hcl
terraform {
  required_providers {
    opennebula = {
      source  = "ig1.com/ig1/opennebula"
      version = "~> 0.1"
    }
  }
}

provider "opennebula" {
  # Configuration options
  endpoint      = "https://www.tata.net"
  flow_endpoint = "titi"
  username      = "toto"
  password      = "tutu"
}

resource "opennebula_group" "example" {
  // ...
}
```

```shell
$ terraform plan

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # opennebula_group.example will be created
  + resource "opennebula_group" "example" {
      // ...
    }

Plan: 1 to add, 0 to change, 0 to destroy.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
```

### PR DESCRIPTION
**Pull request by treywelsh**
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for PR followers and do not help prioritize the request

### Description

Fail on empty credentials.

### References

Close #283

### Checklist

- [ ] I have created an issue and I have mentioned it in `References`
- [ ] My code follows the style guidelines of this project (use `go fmt`)
- [ ] My changes generate no new warnings or errors
- [ ] I have updated the unit tests and they pass succesfuly
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation (if needed)
- [ ] I have updated the changelog file

### CODE REVIEW

**Review by frousselet — COMMENTED** [review]
Tested on my side. Now shows error as expected when using wrong parameters:

```shell
$ terraform plan

╷
│ Error: GOCA client error [REQUEST_HTTP]: http make request: Post "https://www.tata.net": dial tcp: lookup www.tata.net: no such host
│
│   with provider["ig1.com/ig1/opennebula"],
│   on main.tf line 10, in provider "opennebula":
│   10: provider "opennebula" {
│
╵
```

Missing changelog and we are good to merge.

File: opennebula/provider.go [hunk 1]
```diff
@@ -72,30 +75,57 @@ func Provider() *schema.Provider {
...
+
+	version, err := goca.NewController(oneClient).SystemVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[INFO] OpenNebula version: %s", version)
+
+	flowEndpoint, ok := d.GetOk("flow_endpoint")
+	if ok {
```
Comment (frousselet) [inline opennebula/provider.go:112]: What does happen if not `ok`?

**Review by frousselet — CHANGES_REQUESTED** [review]
- [x] Please add changes to CHANGELOG.md

File: opennebula/provider.go (hunk 1 above)
Comment (treywelsh) [reply opennebula/provider.go:112]: The flow configuration is optional in the provider: oneflow is an optional service to configure in OpenNebula.
If we don't provider the flow endpoint then the provider won't be able to work with flow resources (the service resources).

So here, if not ok, the provider will get a controller that isn't able to work with flow resources.

**Review by frousselet — APPROVED** [review]

**Pull request merged**